swingset/src/main/java/com/nqadmin/swingset/SSBaseComboBox.java

```diff
@@ -111,30 +112,73 @@
 	private static final long serialVersionUID = 1L;
 
 	/**
-	 * Listener(s) for the component's value used to propagate changes back to bound
-	 * database column.
+	 * Listener(s) for the component's value used to propagate changes back to the rowset in certain instances.
+	 * <p>
+	 * NOTE: No guaranty whether ActionListener or FocusLister will be called first when focus is lost.
 	 */
 	protected class SSBaseComboBoxListener implements ActionListener, Serializable
 	{
 		private static final long serialVersionUID = -3131533966245488092L;
 
-		/** {@inheritDoc} */
+		/**
+		 * For JComboBox ActionListener and ItemListener are very similar.
+		 * 
+		 * {@inheritDoc} */
 		@Override
 		public void actionPerformed(final ActionEvent ae) {
 
-			removeRowSetListener();
-
-			M mapping = getSelectedMapping();
-
-			if (mapping == null) {
-				setBoundColumnText(null);
-				logger.debug(() -> String.format("%s: Setting to null.", getColumnForLog()));
-			} else {
-				setBoundColumnText(String.valueOf(mapping));
-				logger.debug(String.format("%s: Setting to %s.",  getColumnForLog(), mapping));
+			// If this is a combo navigator, SSSyncManager will have it's own listeners.
+			// This is just for keeping a bound column in sync.
+			//
+			// Could be combined with next block, but keeping them separate for debugging.
+			// May be able to eliminate actionListenerNoUpdate when GlazedLists fully supports STRICT/CONTAINS
+			if (getBoundColumnName() == null) {
+				logger.debug("{}: Action Listener returning. No bound column.", () -> getColumnForLog());
+				return;
 			}
```

Comment:
How about
```
if (isComboBoxNavigator()) {
```

Reply:
Agree on isComboBoxNavigator(). Pushed an update. A navigator will have it's own listener with possible calls to setSelectedItem(). I think it's best to cleanly exit the method for a navigator as I'm not sure if the two listeners might compete. It's possible this method could mostly be reduced to updateRowset() once STRICT/CONTAINS is resolved so we won't have the Action Listeners making additional calls to setSelectedItem(). Feel safer with the conditional return for now.